pkg/apis/hive/v1/agent/installstrategy.go

```diff
@@ -0,0 +1,84 @@
+package agent
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// InstallStrategy is the install strategy configuration for provisioning a cluster with the
+// Agent based assisted installer.
+type InstallStrategy struct {
+	// Networking is the configuration for the pod network provider in
+	// the cluster.
+	Networking Networking `json:"networking"`
```

Comment:
why is networking part of the strategy? to me it looks like a platform configuration?

can you add a new lines describing the difference in strategy and platform somewhere?

Reply:
My thinking was that Networking is likely to be used for any agent based install strategy, and isn't bare metal specific. I actually really wonder if this shouldn't be in the ClusterDeployment itself, to move towards what we talked about a few weeks ago where we might make provising an install config optional, and generate one for you if you don't. We have the option to do that given this is behind an alpha gate, but it did seem specific to the use of the agent installer rather than the agent metal platform. I'm not sure how to convey that in comments in a way that's relevant to an API user but I'll give it a shot.